drivers/interrupt_controller/intc_plic.c

```diff
@@ -115,7 +120,8 @@ static inline const struct device *get_plic_dev_from_irq(uint32_t irq)
 static int riscv_plic_is_edge_irq(const struct device *dev, uint32_t local_irq)
 {
 	const struct plic_config *config = dev->config;
-	mem_addr_t trig_addr = config->trig + local_irq_to_reg_offset(local_irq);
```

Comment:
Shouldn't we just fix `local_irq_to_reg_offset()` in this PR? Offset is in bytes, so we're basically leaving a land mine for whoever hits this problem next time.

Also, kind of wondering how that even passed CI... we have tests, right? Is it a build-only test? Would maybe explain  how `local_irq_to_reg_offset()` was somehow skipped.

Reply:
I think what @ycsin meant is that `reg_offset` is the offset expressed in the number of registers.

As for tests, I'm not sure if these run in CI, but on the QEMU virt platforms the uart interrupt number is small enough such that the calculated offset was (correctly) 0, so the lacking multiplication was of course irrelevant.

Reply:
Maybe a misnomer - reg_index would be better in the case.